dae_testing/homework_example/HW8_3a.py

```diff
@@ -39,24 +72,55 @@ def _x3dot(m, t):
         return m.dx3dt[t] == m.x1[t]**2 + m.x2[t]**2 + 0.005*m.u[t]**2
     m.x3dot = Constraint(m.t, rule=_x3dot)
 
-    def _init(m):
-        yield m.x1[0] == 0
-        yield m.x2[0] == -1
-        yield m.x3[0] == 0
-    m.init_conditions = ConstraintList(rule=_init)
-
-    return m
+#Constraint_definitions_old doesn't ignore constraints at 
+#non-collocation discretization points
 
 
 def discretize_model(
         m,
         method="dae.collocation",
         scheme="LAGRANGE-RADAU",
-        nfe=10,
-        ncp=5,
+        nfe= 2,
+        ncp=3,
         ):
     discretizer = TransformationFactory(method)
     discretizer.apply_to(m, nfe=nfe, ncp=ncp, scheme=scheme)
+    
+
+def discretization_points(m):
+    return m.t.get_finite_elements(), \
+        m.t.get_discretization_info()['tau_points']
+
+#m.t.get_finite_elements = These are all discretization points 
+#m.t.tau_pts = These are roots of polynomials including 0 
+#as an additional element
+
+
+def convert_taupts_to_cp(m, nfe):
+    h = m.tf/nfe
+    cp = []
+    for i in m.disc_pts[:-1] :
+        for j in m.col_roots[1:]:
+            cp.append(i + h*j)
+    return cp
+
+#These convert the polynomial roots to collocation points between 
+#each of the finite element boundaries.
+
+def non_coll_dicretization_pts(m):
+    round_disc_pts = []
+    round_cp = []
+    for i in m.disc_pts: 
+        round_disc_pts.append(round(i, 4))
+    for i in m.cp:
+        round_cp.append(round(i,4))
+
+    non_coll_disc_pts = list(set(round_disc_pts) - set(round_cp))
+    return non_coll_disc_pts
+
+#This function returns the dicretization points which are not collocation points
+#If we don't round the items in the list we have problems with list comparison. 
+#0.3333 is not same as 0.3333333

```

Comment:
I think I made a comment like this elsewhere, but rounding here is not safe, as Pyomo DAE could change how it rounds without warning. Ideally, we would construct `non_coll_disc_pts` from "exact replicas" of the points in the continuous set. If you need to get the exact points in the continuous set from some "hand-calculated" approximation, you can do it with the `ContinuousSet.find_nearest_index` method, which uses binary search.